src/Server/modules/USPlacesManager.js

```diff
@@ -0,0 +1,325 @@
+const dataClassesModule = require('../modules/DataClasses.js');
+const readCSVModule = require('../modules/ReadCSV.js');
+const writeCSVModule = require('../modules/WriteCSV.js');
+
+var allUSPlaces = [];
+
+function InitializeAllUSPlace(){
+    allUSPlaces = readCSVModule.LoadUSData();
+}// read allUSdata and put into array
+
+function AddUSData(county, state, date, tod) {
+    
+    if(allUSPlaces == undefined){
+        InitializeAllUSPlace();
+    }
+
+	var errormsg = 'no error';
+	if (tod == 1) {
+		for (var i = 0; i < allUSPlaces.length; ++i) {
+			if (county == allUSPlaces[i].county && state == allUSPlaces[i].state) {
+				for (var key in allUSPlaces[i].currentNumConfirmed) {
+					var temp_date = new Date(key);
+					var d =
+						temp_date.getMonth() +
+						1 +
+						'/' +
+						temp_date.getDate() +
+						'/' +
+						temp_date.getFullYear();
+					if (d == date) {
+						console.log('ERROOOOOOOR');
+						//return result;
+						errormsg = 'date exists';
+						return errormsg;
+					}
+				}
+				var temp_date = helper.stringToDate(date);
+				allUSPlaces[i].currentNumConfirmed[temp_date] = number;
+				break;
+			} else if (i == allUSPlaces.length - 1) {
+				// if its the last element & prog didn't go into the 1st if -> place doesn't exist
+				errormsg = 'wrong place';
+			}
+		}
+	} else if (tod == 2) {
+		for (var i = 0; i < allUSPlaces.length; ++i) {
+			if (county == allUSPlaces[i].county && state == allUSPlaces[i].state) {
+				for (var key in allUSPlaces[i].currentNumDeaths) {
+					var temp_date = new Date(key);
+					var d =
+						temp_date.getMonth() +
+						1 +
+						'/' +
+						temp_date.getDate() +
+						'/' +
+						temp_date.getFullYear();
+					if (d == date) {
+						console.log('ERROOOOOOOR');
+						errormsg = 'date exists';
+						return errormsg;
+					}
+				}
+				var temp_date = helper.stringToDate(date);
+				allUSPlaces[i].currentNumConfirmed[temp_date] = number;
+				break;
+			} else if (i == allUSPlaces.length - 1) {
+				// if its the last element & prog didn't go into the 1st if -> place doesn't exist
+				errormsg = 'wrong place';
+				break; //just in case
+			}
+		}
+	}
+	
+
+	return errormsg;
```

Comment:
When you add a data, you need to update both currentNumConfirmed and numConfirmed.

numConfirmed: dictionary of <Date, integer>. It stores something like { {1/20/20, 4}, {1/21/20, 6}, {1/22/20, 10}, ... }
currentNumConfirmed: integer. It stores the number of cases for the most recent date in the numConfirmed.
(For example, if numConfirmed = { {1/20/20, 4}, {1/21/20, 6}, {1/22/20, 10}, {1/23/20, 21}}, then currentNumConfirmed = 21, because 1/23/20 is the most recent date of all dates in numConfirmed.

Same thing for numDeaths and currentNumDeaths.